neo/Consensus/ConsensusContext.cs

```diff
@@ -30,12 +30,18 @@ internal class ConsensusContext : IConsensusContext
         public UInt256[] TransactionHashes { get; set; }
         public Dictionary<UInt256, Transaction> Transactions { get; set; }
         public UInt256[] Preparations { get; set; }
+        public byte[][] PreparationWitnessInvocationScripts { get; set; }
+        public uint[] PreparationTimestamps { get; set; }
         public byte[][] Commits { get; set; }
         public byte[] ExpectedView { get; set; }
+        public byte[][] ChangeViewWitnessInvocationScripts { get; set; }
+        public uint[] ChangeViewTimestamps { get; set; }
+        public byte[] OriginalChangeViewNumbers { get; set; }
         private Snapshot snapshot;
         private KeyPair keyPair;
         private readonly Wallet wallet;
 
+        public int F => (Validators.Length - 1) / 3;
         public int M => Validators.Length - (Validators.Length - 1) / 3;
```

Comment:
```suggestion
        public int M => Validators.Length - F;
```

Reply:
I thought about doing this actually and had it this way for a minute, but I don't think it will be inlined and will incur an extra method call every time it gets `M`. I guess the performance impact is so minor compared to all the other things we are doing though that maybe just for readability, I will make this change.